### ISSUE TITLE
Does GRBL use line numbers in any way?

### ISSUE THREAD

**Issue by fedekrum** [grbl/grbl#1534]
I am trying to implement a remote offline controller.
As in this device memory is critical for me, I am trying to parse the gcode and remove spaces, comments, empty lines.
Y was wondering if I can safely remove N1234 characters at the beginning of the lines for GRBL specific code.

Thanks

**Comment by jahnj0584** [comment]
Yes. It's only for user reference. Most Gcode editors have a button to
add/remove line numbers.

All lathe codes i've seen dont use it because you never restart midway
through an operation and solely go by tool number.

**Comment by fedekrum** [comment]
Thanks @jahnj0584  !!
Any other thing that can reduce code size that you can think off?

**Comment by PicEngraver** [comment]
Remove repeating modal commands such as G1, etc.

**Comment by Norwegian-Gadgetman** [comment]
Consider the accuracy of your machine, and how many digits of accuracy is in the commands.   
A G1 X15.12345 (in millimeters) is a waste of digits if your machine isn't capable of that accuracy.  
Also, many tools will generate positions for ALL axis even if it doesn't need them. 
If your tool is at X20, Y15,,Z10, a G1 X20, Y15, Z0 can be replaced with just the G1 Z0. 
(Assuming absolute positioning, of course. In Relative positioning any axis that reads o can be removed)

**Comment by connectable2** [comment]
fedekrum's point is well-made regarding size constraints. I generate most of my gcode using python, so a simple switch turns line numbering on or off. Usually I leave them in, so that simple editors show them, otherwise you are lost at sea..
